tipg/db.py

```diff
@@ -11,14 +12,37 @@
 from fastapi import FastAPI
 
 
-async def con_init(conn):
+async def con_init(
+    conn,
+    settings: Optional[PostgresSettings] = None,
```

Comment:
This will make customization a bit hard, because we won't be able to pass any settings to it

Reply:
In order to use the temp schema, we need to make sure that the functions_directory scripts run on each and every connection. Doing it when creating the pool does not work.

Reply:
Sure, I'll think about something 

```python
from pydantic import BaseModel

class Conn(BaseModel):

    settings: PostgresSettings

    async def __call__(self, conn)
        await conn.set_type_codec(
            "json", encoder=orjson.dumps, decoder=orjson.loads, schema="pg_catalog"
        )
        await conn.set_type_codec(
            "jsonb", encoder=orjson.dumps, decoder=orjson.loads, schema="pg_catalog"
        )
        await conn.execute(
            """
                SELECT set_config(
                    'search_path',
                    'pg_temp,' || current_setting('search_path', false),
                    false
                    );
                """
        )
        if (
            self.settings.tipg_functions_directory
            and Path(self.settings.tipg_functions_directory).exists()
        ):
            for sqlfile in Path(self.settings.tipg_functions_directory).glob("**/*.sql"):
                await conn.execute(sqlfile.read_text())
        
        dbcatalogsql = Path(__package__ / "sql" / "dbcatalog.sql")
        
        await conn.execute(dbcatalogsql.read_text())
```

Reply:
but let me think about this more